plugins/azure-devops/src/components/EntityPageAzureGitTags/EntityPageAzureGitTags.tsx

```diff
@@ -14,7 +14,29 @@
  * limitations under the License.
  */
 
+import { azureDevOpsGitTagReadPermission } from '@backstage/plugin-azure-devops-common';
 import { GitTagTable } from '../GitTagTable/GitTagTable';
 import React from 'react';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { RequirePermission } from '@backstage/plugin-permission-react';
+import { EmptyState } from '@backstage/core-components';
 
-export const EntityPageAzureGitTags = () => <GitTagTable />;
+export const EntityPageAzureGitTags = () => {
+  const { entity } = useEntity();
+  return (
+    <RequirePermission
+      permission={azureDevOpsGitTagReadPermission}
+      resourceRef={stringifyEntityRef(entity)}
+      errorPage={
+        <EmptyState
+          missing="data"
+          title="No Tags to show"
+          description="You are not authorized!"
```

Comment:
I'm only going to leave one comment, but this would apply for all the `EmptyState` usage. I'm not a Security expert but I was under the impression that it's best not to share "not authorized" messages but use a more generic not found. I'm just not sure the right messaging here, yet.

I do think using `EmptyState` makes sense as we the default `errorPage` would potentially look really bad.

Reply:
@vinzscam do you have any suggestions for this?

Reply:
maybe we can just hide the component?

`errorPage={null}`

Reply:
@vinzscam @awanlin 
Yes it will be an Empty page if the permission failed not only `Git Tags` Entity Page on `Pipelines` and `PRs` too.

What is your suggestion, will respond with `Not authorized` Card or leave it to default error message in `Warning Panel` or the with `Empty Page` ?

Reply:
I'm just doing a review now, going to pull the code down and play around with it. Will follow up with my thoughts.

Reply:
From my testing this is fine for the Pipelines, Pull Request Table, and Git Tags. The Readme card and the Pull Request Dashboard on the other hand don't work, you hit the `NotAllowedError`

The Readme card is due to hitting this line first: https://github.com/backstage/backstage/blob/ca252dd722a795109083c00af4ea42233df7b94f/plugins/azure-devops/src/components/ReadmeCard/ReadmeCard.tsx#L98 Maybe we need to wrap it in an `EntityCardReadme`?

For the Pull Request Dashboard I think you just need to warp this with `RequirePermission`: :https://github.com/backstage/backstage/blob/ca252dd722a795109083c00af4ea42233df7b94f/plugins/azure-devops/src/components/PullRequestsPage/PullRequestsPage.tsx#L101

Reply:
@awanlin 

https://github.com/backstage/backstage/pull/22591#discussion_r1484790798
- `ReadmeCard` yes you are right. I was using use like below, before using `RequirePermission` during change, totally forget the hook before the component.
  ```ts
  if (isAuthorizationError(error)) {
      return (
        <EmptyState
          title="No README available for this entity"
          missing="field"
          description="You are not authorized!"
        />
      );
    }
  ``` 
   I will create `EntityCardReadme` and wrap with `RequirePermission`
- `PullRequestDashbaord` Since it is a Basic Permission. I let the adopter add it on the Route element as mentioned here https://backstage.io/docs/permissions/frontend-integration. Ofcourse I didn't document it. I will update the documentation with required details.